lab-jessica/lib/storage.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+const del = require('del');
+const mkdirp = require('mkdirp');
+const Promise = require('bluebird');
+const fs = Promise.promisifyAll(require('fs'), {suffix: 'Prom'});
+
+module.exports = exports = {};
+
+exports.createItem = function(schemaName, item) {
+  if (!schemaName) return Promise.reject(new Error('expected schemaName'));
+  if (!item) return Promise.reject(new Error('expected item'));
+  let json = JSON.stringify(item);
+  mkdirp(`${__dirname}/../data/${schemaName}/`, function() {
+    fs.writeFileProm(`${__dirname}/../data/${schemaName}/${item.id}.json`, json)
+    .then( () => item)
+    .catch( err => Promise.reject(err));
+  });
+};
+
+exports.fetchItem = function(schemaName, id) {
+  if (!schemaName) return Promise.reject(new Error('expected schemaName'));
+  if (!id) return Promise.reject(new Error('expected id'));
+
+  return fs.readFileProm(`${__dirname}/../data/${schemaName}/${id}.json`)
+  .then(data => {
+    try {
+      let item = JSON.parse(data.toString());
+      return item;
+    } catch (err) {
+      return Promise.reject(err);
+    }
+  })
+  .catch(err => Promise.reject(err));
+};
+
+// exports.fetchAll = function(schemaName) {
+//   if (!schemaName) return Promise.reject(new Error('expected schemaName'));
+//
+//   fs.readdirProm(`${__dirname}/../data/${schemaName}`)
```

Comment:
This should return a list of file names, so you can iterate over each of those and call fetchItem on each to return the whole contents of all objects.